### PR TITLE
move plumbing from top level package to plumbing

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

**Comment by smola** [comment]
Maybe we should merge this PR without squash.

**Comment by mcuadros** [comment]
If you don't mind, can you include in this PR, rename some files in the root of the project to singular?

File: common_test.go [hunk 1]
```diff
@@ -9,13 +9,15 @@ import (
 	"gopkg.in/src-d/go-git.v4/plumbing/transport/client"
 
 	. "gopkg.in/check.v1"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
```
Comment (ajnavarro) [inline common_test.go]: Bad import order.

File: cshared/tree_cshared.go [hunk 2]
```diff
@@ -163,7 +163,7 @@ func c_TreeWalker_Close(tw uint64) {
 	if !ok {
 		return
 	}
-	walker := obj.(*git.TreeIter)
+	walker := obj.(*object.TreeIter)
 	walker.Close()
 }
 */
```
Comment (ajnavarro) [inline cshared/tree_cshared.go]: All this file is commented? maybe we can delete it in this refactor.

File: plumbing/object/tree_test.go [hunk 3]
```diff
@@ -1418,7 +1419,7 @@ func (s *TreeSuite) TestTreeDecodeReadBug(c *C) {
 			},
 		},
 		Hash: plumbing.Hash{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
```
Comment (ajnavarro) [inline plumbing/object/tree_test.go]: ZeroHash? (maybe is not the correct PR to change it)

File: references_test.go [hunk 4]
```diff
@@ -8,6 +8,7 @@ import (
 	"gopkg.in/src-d/go-git.v4/plumbing"
 
 	. "gopkg.in/check.v1"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
```
Comment (ajnavarro) [inline references_test.go]: Bad import order.

**Comment by smola** [comment]
@mcuadros Done. Except for references.go (which is for a function called References, which we might remove/move anyway) and options.go (which actually defines Options objects).

File: cshared/tree_cshared.go (hunk 2 above)
Comment (smola) [inline cshared/tree_cshared.go]: Yep. cshared package is half-commented-half-broken until we stabilize the API.

**Comment by mcuadros** [comment]
fuck, I merged doing squash, sorry

**Pull request merged**